pkg/scalers/kafka_scaler.go

```diff
@@ -57,6 +65,7 @@ const (
 	lagThresholdMetricName   = "lagThreshold"
 	kafkaMetricType          = "External"
 	defaultKafkaLagThreshold = 10
+	defaultOffsetReset       = earliest
```

Comment:
in Kafka the default is "latest", why are you setting "earliest" out of curiosity?

Reply:
Yes, you are right on the default in kafka. I set latest as a default because, right now, if you create a kafka scaler, it will behave as if we set earliest, if no offset is committed. So I would not break the current behavior for anyone upgrading from a previous version and not seeing the option.
But maybe, since this is a new major version of Keda, it could be safer to be coherent with Kafka defaults. 
Let me know what you think, I can change it accordingly to the decision.

Reply:
yes I would agree, let's see what @zroubalik thinks.

Reply:
I agree, it makes sense to set it to Kafka default, a new major release is a great fit for a change like this. We could add a small note to the docs about it.

Reply:
I've done the change and also changed the docs, let me know if it needs further clarification there.